### PR TITLE
add mock-tikv client

### PR DESCRIPTION
**Pull request by disksing**
Wrap mock-tikv's HTTP API. Make it easy to create a mock cluster dynamically.

Signed-off-by: disksing <i@disksing.com>

### CODE REVIEW

File: mocktikv/cluster.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	return c.pdAddrs
+}
+
+// Close releases the mock cluster.
+func (c *Cluster) Close() error {
+	req, err := http.NewRequest("DELETE", c.mockServer+"/mock-tikv/api/v1/clusters", nil)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	_, err = http.DefaultClient.Do(req)
```
Comment (rleungx) [inline mocktikv/cluster.go]: I think it's better not to use the default HTTP client.

Comment (disksing) [reply mocktikv/cluster.go]: ok

**Comment by rleungx** [comment]
Do we need to merge tikv/mock-tikv#1 firstly?

**Comment by disksing** [comment]
@rleungx I'm ok. You can review that if you feel it necessary :smiley:

**Comment by disksing** [comment]
@rleungx mock-tikv PR is merged.

**Review by rleungx — APPROVED** [review]

**Pull request merged**